src/components/Checkbox.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @typedef CheckboxBaseProps
+ * @prop {string} name - The `name` of the checkbox.
+ * @prop {import('preact').Ref<HTMLInputElement>} [inputRef]
```

Comment:
Possible to document this a little more? What's the intent of this? I think I get it but want to feel more certain.